packages/gatsby/cache-dir/loader.js

```diff
@@ -353,7 +353,12 @@ const createComponentUrls = componentChunkName =>
 export class ProdLoader extends BaseLoader {
   constructor(asyncRequires, matchPaths) {
     const loadComponent = chunkName =>
-      asyncRequires.components[chunkName]().then(preferDefault)
+      asyncRequires.components[chunkName]()
+        .then(preferDefault)
+        .catch(() => {
+          console.log(`failed to load component but not an unhandled exception`)
+          return null
```

Comment:
not sure if best place to catch - but we do handle cases when this resolves to `null` in loader - so this was quickest way to "fix" that